pkg/apis/pipeline/v1beta1/result_validation.go

```diff
@@ -17,22 +17,21 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/tektoncd/pipeline/pkg/apis/config"
 	"knative.dev/pkg/apis"
 )
 
 // Validate implements apis.Validatable
-func (tr TaskResult) Validate(_ context.Context) *apis.FieldError {
+func (tr TaskResult) Validate(ctx context.Context) (errs *apis.FieldError) {
 	if !resultNameFormatRegex.MatchString(tr.Name) {
 		return apis.ErrInvalidKeyName(tr.Name, "name", fmt.Sprintf("Name must consist of alphanumeric characters, '-', '_', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my-name',  or 'my_name', regex used for validation is '%s')", ResultNameFormat))
 	}
-	// Validate the result type
-	validType := false
-	for _, allowedType := range AllResultsTypes {
-		if tr.Type == allowedType {
-			validType = true
-		}
+	// Array and Object is alpha feature
+	if tr.Type == ResultsTypeArray || tr.Type == ResultsTypeObject {
+		return errs.Also(ValidateEnabledAPIFields(ctx, "results type", config.AlphaAPIFields))
 	}
-	if !validType {
+
+	if tr.Type != ResultsTypeString {
```

Comment:
wouldn't this condition cause failure for an array and a map results? 

The previous if condition is validating that the type cannot be array/map without `alpha` feature flag. 

So, lets say, the alpha feature flag is set and result is array, this if condition will fail and cause validation error, right? 🤔 

Also, not a blocker for this PR but please add a unit test `result_validation_test.go`

Reply:
Thanks for the comment! 
If the type is `ResultsTypeArray` or `ResultsTypeObject`, we will directly return this check and won't reach the following code. So they won't fail validation if  alpha feature flag is set. 
```
return errs.Also(ValidateEnabledAPIFields(ctx, "results type", config.AlphaAPIFields))
```

Reply:
This function is tested in `task_validation_test.go`,  but I can add separate test for this func for sure! Maybe save it for another PR?

Reply:
thank you @Yongxuanzhang I commented a little early 🙃  thought there was more to this but apparently its only checking the types.